module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -537,26 +554,23 @@ namespace module::platform::darwin {
                 /************************************************
                  *                  Kinematics Horizon          *
                  ************************************************/
-                Eigen::Affine3d Rwt(sensors->Htw.inverse());
-                // remove translation components from the transform
-                Rwt.translation() = Eigen::Vector3d::Zero();
-                Eigen::Affine3d Rgt(
-                    Eigen::AngleAxisd(-Rwt.rotation().eulerAngles(0, 1, 2).z(), Eigen::Vector3d::UnitZ()) * Rwt);
-                // sensors->Hgt : Mat size [4x4] (default identity)
-                // createRotationZ : Mat size [3x3]
-                // Rwt : Mat size [3x3]
+                // Extract the inverse of the rotation component of Htw
+                const Eigen::Matrix3d Rwt(sensors->Htw.topLeftCorner<3, 3>().transpose());
+                // We remove the yaw by making an angleaxis which has just the negative yaw,
+                // then multiplying it back in, taking the yaw away
+                const Eigen::AngleAxisd Rwt_negative_yaw(-Rwt.eulerAngles(0, 1, 2).z(), Eigen::Vector3d::UnitZ());
+                const Eigen::Affine3d Rgt(Rwt_negative_yaw * Rwt);
+
                 sensors->Hgt = Rgt.matrix();
-                auto Htc     = sensors->Htx[ServoID::HEAD_PITCH];
 
-                // Get torso to world transform
-                Eigen::Affine3d yawlessWorldInvR(
+                // Get torso to ground transform (then do nothing with it????)
+                Eigen::AngleAxisd yawlessWorldInvR(
+                    // Also removing the yaw in the same way
                     Eigen::AngleAxisd(-Hwt.rotation().eulerAngles(0, 1, 2).z(), Eigen::Vector3d::UnitZ())
                     * Hwt.rotation());
-                Eigen::Affine3d Hgt(Hwt);
+                Eigen::Affine3d Hgt;
                 Hgt.translation() = Eigen::Vector3d(0, 0, Hwt.translation().z());
-                Hgt.linear()      = yawlessWorldInvR.linear();
-                sensors->Hgc      = Hgt * Htc;  // Rwt * Rth
-
+                Hgt.linear()      = yawlessWorldInvR.toRotationMatrix();
```

Comment:
We calculate a different `Hgt` to the one we just put in the `sensors` message, then do nothing with it. This seems wrong.

Reply:
This is two things sharing the same name, This Hgt is an very old thing that is hopefully not used anywhere anymore

Reply:
I think @Bidski did that at last RoboCup

Reply:
so I should delete everything from the `kinematics horizon` block comment up to but not including the `emit(std::move(sensors))`?

Reply:
I have deleted the stuff which wasn't added to the `sensors` message which seemed to serve zero purpose.